background.js

```diff
@@ -31,18 +32,21 @@ function resetState(tabId) {
   };
 }
 
-function bufferData(tabId, message) {
-  var tabData = data[tabId],
-      scope;
-
+function setIsHintFlag(message) {
```

Comment:
It updates the message, which i wouldn't expect a function called bufferData to do.